Apps/Src/SendTritium.c

```diff
@@ -0,0 +1,426 @@
+        state = BRAKE_STATE;
+    }else if(regenToggle){
+        cruiseEnable = false;
+        state = NORMAL_DRIVE;
+    }
+}
+
+/**
+ * Coasting Cruise State. We do not want to utilize motor braking in cruise control mode.
+ * Coast the motor (do not motor brake) if we want to slow down; instead, stop putting power
+ * into the motor and let the car slow down naturally.
+*/
+void CoastingCruiseHandler(){
+    currentSetpoint = 0;
+}
+
+void CoastingCruiseDecider(){
+    if(MotorVelocity_Get() <= velocitySetpoint){
+        state = POWERED_CRUISE;
+    }else if(!cruiseEnable){
+        state = NORMAL_DRIVE;
+    }else if(brakePedalPercent >= NORMAL_BRAKE_THRESHOLD){
+        state = BRAKE_STATE;
+    }else if(regenToggle){
+        cruiseEnable = false;
+        state = NORMAL_DRIVE;
+    }
+}
+
+/**
+ * Brake State. Brake the car if the brake pedal is pressed.
+*/
+void BrakeHandler(){
+    currentSetpoint = 0;
+    velocitySetpoint = 0;
+    cruiseEnable = 0;
+    cruiseSet = 0;
+    brakelightState = true;
+}
+
+void BrakeDecider(){
+    if(brakePedalPercent < NORMAL_BRAKE_THRESHOLD){
+        state = NORMAL_DRIVE;
+        brakelightState = false;
+    }
+}
+
+/**
+ * One Pedal Drive State. When in one pedal drive, if the accelerator percentage is lower
+ * than ONEPEDAL_BRAKE_THRESHOLD, the car will utilize motor braking to slow down. If accelerator
+ * percentage is in the neutral zone, the car will coast. If accelerator percentage is above
+ * the NEUTRAL_THRESHOLD, the car will accelerate as normal.
+*/
+void OnePedalDriveHandler(){
+    if(accelPedalPercent < ONEPEDAL_BRAKE_THRESHOLD){
+        // Motor brake
+        currentSetpoint = percentToFloat(((ONEPEDAL_BRAKE_THRESHOLD - accelPedalPercent)*100)/ONEPEDAL_BRAKE_THRESHOLD);
+        velocitySetpoint = 0;
+        brakelightState = true;
+    }else if(accelPedalPercent < NEUTRAL_THRESHOLD  && accelPedalPercent >= ONEPEDAL_BRAKE_THRESHOLD){
+        // Coast
+        currentSetpoint = 0;
+        brakelightState = false;
+    }else if(accelPedalPercent >= NEUTRAL_THRESHOLD){
+        // Accelerate
+        currentSetpoint = percentToFloat(((accelPedalPercent - NEUTRAL_THRESHOLD)*100)/(100-NEUTRAL_THRESHOLD));
+        brakelightState = false;
+    }
+}
+
+void OnePedalDriveDecider(){
+    if(!regenToggle || !ChargeEnable || cruiseEnable){
+        regenToggle = 0;
+        state = NORMAL_DRIVE;
+    }else if(brakePedalPercent >= NORMAL_BRAKE_THRESHOLD){
+        state = BRAKE_STATE;
+    }
+}
+
+
+void AccelerateCruiseHandler(){
+    if(reverseSwitch && !forwardSwitch){
+        velocitySetpoint = -MAX_VELOCITY;
+        currentSetpoint = percentToFloat(accelPedalPercent);
+    }else if(!reverseSwitch && forwardSwitch){
+        velocitySetpoint = MAX_VELOCITY;
+        currentSetpoint = percentToFloat(accelPedalPercent);
+    }else if(!reverseSwitch && !forwardSwitch){
+        currentSetpoint = 0;
+    }
+}
+
+void AccelerateCruiseDecider(){
+    if(accelPedalPercent == 0){
```

Comment:
We may need to define a threshold macro for accel pedal as well, I'm worried that it might float a little bit and be at like 0.1 or smth.